scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,246 @@
+        tagmanifest_name = "tagmanifest-sha256.txt"
+        archive_name = f"{working_id}.tar.gz"
+        s3_upload_prefix = "born-digital/archivematica-uuid-update"
+
+        if not os.path.exists(working_folder):
+            os.makedirs(working_folder)
+
+        prefix_patterns = ["bagit.txt", "bag-info.txt", "manifest-", "tagmanifest-"]
+
+        def _write_fetch_file():
+            def _create_fetch_line(file):
+                return f"{path_prefix}/{file['path']}\t{file['size']}\t{file['name']}\n"
+
+            files = document["files"]
+
+            fetch_file = open(f"{working_folder}/fetch.txt", "w")
+            for file in files:
+                fetch_file.write(_create_fetch_line(file))
+            fetch_file.close()
+
+        def _filter_bag_files(prefix):
+            response = storage_s3_client.list_objects_v2(
+                Bucket=bucket, Prefix=f"{path}/{version}/{prefix}"
+            )
+
+            if "Contents" in response:
+                return [content["Key"] for content in response["Contents"]]
+            else:
+                return []
+
+        def _check_expected_prefixes_exist(file_locations):
+            for prefix in prefix_patterns:
+                found_match = False
+
+                for location in file_locations:
+                    if prefix in location:
+                        found_match = True
+
+                if not found_match:
+                    raise RuntimeError(f"Missing any files matching prefix: {prefix}")
+
+        def _get_bag_files():
+            file_locations = []
+            for prefix in prefix_patterns:
+                file_locations = file_locations + _filter_bag_files(prefix)
+
+            _check_expected_prefixes_exist(file_locations)
+
+            for location in file_locations:
+                filename = location.split("/")[-1]
+                save_path = f"{working_folder}/{filename}"
+                storage_s3_client.download_file(bucket, location, save_path)
+
+        def _compress_bag():
+            shutil.make_archive(working_folder, "gztar", working_folder)
+            shutil.rmtree(working_folder, ignore_errors=True)
+
+        def _generate_checksum(file_location):
+            sha256_hash = hashlib.sha256()
+
+            with open(file_location, "rb") as f:
+                for byte_block in iter(lambda: f.read(4096), b""):
+                    sha256_hash.update(byte_block)
+
+                return sha256_hash.hexdigest()
+
+        def _generate_updated_checksums():
+            files_in_need_of_update = ["bag-info.txt", "fetch.txt"]
+
+            return {
+                filename: _generate_checksum(f"{working_folder}/{filename}")
+                for filename in files_in_need_of_update
+            }
+
+        def _load_existing_checksums():
+            tag_manifest = {}
+
+            with open(f"{working_folder}/{tagmanifest_name}") as fp:
+                for _, line in enumerate(fp):
```

Comment:
minor: don't think enumerate adds anything?

```suggestion
                for line in fp:
```